### PR TITLE
Easy hard health

### PR DESCRIPTION
**Pull request by cefong**
Lots of changes:
- added easy/hard health mode, adjustable from main screen
- added tutorial scene that plays immediately after the opening cutscene
- added stationary hazards (spikes, two types of catnip), they're in prefabs and customizable for how much damage they deal
- added updates to player progress so that it automatically calculates the score the player needs to get to pass the level
- added three new cats as walking/patrolling enemies in prefabs
- removed health stuff from flying enemy script since that is in enemy health now
- added dodge mechanic for malakai, press down or s to activate it. You cannot attack while dodging

**Brief Instructions on using various prefabs**
_Malakai_
- attach the health bar to the health bar area (don't mess with the blocking field, that's for player attack to access only)
- attach the progress bar and the door to the player progress script
_EnemyHealth_
- attach malakai as the player object in enemyHealth of any enemy prefab

Basically, just follow any nullreference errors you get in the console and if you need help just ask!

### CODE REVIEW

**Review by balreetg — APPROVED** [review]
OMG This is a amazing!!!
